controllers/mirrorpeer_controller.go

```diff
@@ -508,3 +518,80 @@ func (r *MirrorPeerReconciler) createDRClusters(ctx context.Context, mp *multicl
 	}
 	return nil
 }
+
+func (r *MirrorPeerReconciler) createRoleBindingsForSpoke(ctx context.Context, peer multiclusterv1alpha1.MirrorPeer) error {
+	crb := rbacv1.ClusterRoleBinding{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: spokeClusterRoleBindingName}, &crb)
+	if err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return err
+		}
+	}
+	var subjects []rbacv1.Subject
+	if crb.Subjects != nil {
+		subjects = crb.Subjects
+	}
+
+	// Add users
+	for _, pr := range peer.Spec.Items {
+		sub := getSubjectByPeerRef(pr, "User")
+		if len(subjects) == 0 {
+			subjects = append(subjects, *sub)
+		} else {
+			if !utils.ContainsSubject(subjects, sub) {
+				subjects = append(subjects, *sub)
+			}
+		}
+	}
+
+	// Add groups
+	for _, pr := range peer.Spec.Items {
+		sub := getSubjectByPeerRef(pr, "Group")
+		if len(subjects) == 0 {
+			subjects = append(subjects, *sub)
+		} else {
+			if !utils.ContainsSubject(subjects, sub) {
+				subjects = append(subjects, *sub)
+			}
+		}
+	}
+
+	spokeRoleBinding := rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: spokeClusterRoleBindingName,
```

Comment:
Here, the role binding name is always the same, but you are adding diff subjects for different clusters under the same role binding, is it correct?

Reply:
yes, the idea with this is to give limited permissions to the controller over security resources. The controller will only be able use this resource. Since it is created on hub, all MirrorPeers can use this and add the required permissions for each managed cluster